code/modules/projectiles/projectile/magic.dm

```diff
@@ -158,7 +158,8 @@
 						if("drone")
 							new_mob = new /mob/living/simple_animal/drone(M.loc)
 							var/mob/living/simple_animal/drone/D = new_mob
-							D.update_drone_hack()
+							spawn(50) //this should give us time for the rest of the proc to finish so a mob will be in the drone
```

Comment:
an istype at the bottom is better than a spawn I reckon, anyway, probably up to a maintainer to determine which is nicer.

what we really need is the ability to pass procs/blocks of code to new mobs to run at the end of their initialization.